stm/asyncTaskQueue.mli

```diff
@@ -6,79 +6,129 @@
 (*         *       GNU Lesser General Public License Version 2.1        *)
 (************************************************************************)
 
+(** A status is
+    - `Fresh when XXX
+    - `Old x when XXX
+  *)
 type 'a worker_status = [ `Fresh | `Old of 'a ]
 
+(** The [Task] module type defines an abstract message-processing
+    entity. *)
 module type Task = sig
 
+  (** Identifier of a task, for workers that can process several
+      jobs. Usually an integer or any other unique identifier. *)
   type task
+
+  (** XXX No idea what this does *)
```

Comment:
This is used to describe the data on `Old` worker is responsible of. E.g. a worker that computed an entire proof may hang there, and record in the `competence` that he holds the system states, say, `[1;3;5]` (imagine it did not send back to master these states).

Reply:
What is an `Old` worker; I'm afraid that I don't understand the `worker_status` type yet.